libs/blocks/footer/footer.js

```diff
@@ -1 +1,206 @@
+const ADCHOICE_IMG = '<img class="footer-link-img" loading="lazy" alt="AdChoices icon" src="/libs/blocks/footer/adchoices-small.svg" height="9" width="9">';
+
+class Footer {
+  constructor(body, el) {
+    this.el = el;
+    this.body = body;
+    this.desktop = window.matchMedia('(min-width: 900px)');
+  }
+
+  init = async () => {
+    const wrapper = createTag('div', { class: 'footer-wrapper' });
+
+    const grid = this.decorateGrid();
+    if (grid) {
+      wrapper.append(grid);
+    }
+
+    const infoRow = createTag('div', { class: 'footer-info' });
+    const infoColumnLeft = createTag('div', { class: 'footer-info-column' });
+    const infoColumnRight = createTag('div', { class: 'footer-info-column' });
+
+    const region = await this.decorateRegion();
+    if (region) {
+      infoColumnLeft.append(region);
+      infoRow.classList.add('has-region');
+    }
+
+    const social = this.decorateSocial();
+    if (social) {
+      infoColumnLeft.append(social);
+      infoRow.classList.add('has-social');
+    }
+
+    const privacy = this.decoratePrivacy();
+    if (privacy) {
+      infoColumnRight.append(privacy);
+      infoRow.classList.add('has-privacy');
+    }
+
+    if (infoColumnLeft.hasChildNodes()) {
+      infoRow.append(infoColumnLeft);
+    }
+    if (infoColumnRight.hasChildNodes()) {
+      infoRow.append(infoColumnRight);
+    }
+    if (infoRow.hasChildNodes()) {
+      wrapper.append(infoRow);
+    }
+
+    this.el.append(wrapper);
+  };
+
+  decorateGrid = () => {
+    const gridBlock = this.body.querySelectorAll('div');
+    const footer = document.querySelector('footer');
+    if (!gridBlock) return null;
+    this.desktop.addEventListener('change', this.onMediaChange);
+    const navGrid = createTag('div', { class: 'footer-nav-grid' });
+    const columns = gridBlock;
+    const columnsArray = Array.from(columns);
+    const regionSelectorIndex = columnsArray.findIndex((column) => column.classList.contains('region-selector'));
+    const navCols = columnsArray.slice(0, regionSelectorIndex - 1);
+    navCols.forEach((column) => {
+      if (column) {
```

Comment:
nit: cleaner to early return
```suggestion
      if (!column) return;
```